File: java/07_pgm_thread/src/co/edu/campusucc/poo/MainThreadRunnable.java
package co.edu.campusucc.poo;

import java.util.Random;

import co.edu.campusucc.poo.domain.ThreadRunnable;

public class MainThreadRunnable {

    public static void main(String[] args) {
        ThreadRunnable runnable = new ThreadRunnable();
        runnable.setTimes(new Random().nextInt(50));
        // initializing Thread Object
        Thread thread = new Thread(runnable);
        thread.start();
    }
}

File: c++/07_pgm_thread/src/co/edu/campusucc/poo/MainThreadRunnable.cpp
#include <iostream>
#include <thread>
#include <random>
#include "ThreadRunnable.h"

using namespace std;

int main() {
ThreadRunnable runnable;
runnable.setTimes(rand() % 50); // Generating a random number between 0 and 49
thread threadObj(runnable); // Initializing Thread Object
threadObj.start();
return 0;
}